lib/orbit_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,127 @@
+defmodule OrbitWeb.AuthController do
+  use OrbitWeb, :controller
+  plug(Ueberauth)
+
+  alias OrbitWeb.Auth.Auth
+  require Logger
+
+  def login_page(conn, _params) do
+    if Auth.logged_in_user(conn) do
+      conn
+      |> redirect_login_successful()
+    else
+      conn |> redirect(to: ~p"/auth/keycloak")
+    end
+  end
+
+  def request(conn, _params) do
+    # unrecognized provider
+    # (recognized providers are caught by Ueberauth before this point)
+    conn
+    |> put_status(:bad_request)
+    |> text("Bad Request (unrecognized provider)")
+  end
+
+  defp get_logout_url(conn, auth) do
+    case UeberauthOidcc.initiate_logout_url(auth, %{
+           # We have to generate exactly what Keycloak/EntraID expect for logout redirect
+           post_logout_redirect_uri:
+             URI.parse(url(conn, ~p"/login"))
```

Comment:
**nitpick:** if you're pipelining the rest of this, you should also pull this out (formatting as a suggestion but you probably shouldn't commit it directly because that would break the formatting).

```suggestion
             conn |> url(~p"/login") |> URI.parse()
```